### ISSUE TITLE
`location` object

### ISSUE THREAD

**Issue by kiliankoe** [public-transport/friendly-public-transport-format#19]
The `journey` definition speaks of `location` objects. I'm guessing these reference locations that are neither stops/stations/etc. and are currently still left out of the spec?

**Comment by juliuste** [comment]
You're right! We really need some way to specify the notion of a `location`, e.g. in order to enable people to build address-to-address-routing using FPTF. We just discussed this and propose a new `location` type, which would allow routing between locations/stations/stops and also replace the current, optional `address` and `coordinates` keys in `station` and `stop`. We came up with something like this, we had a slight disagreement about the best solution for saving coordinates though:

```js
{
    type: 'location',
    address: 'Platz der Republik 1, 11011 Berlin', // optional
    name: 'Reichstagsgebäude' // optional,

    // first proposal for coordinates
    coordinates: { // optional, maybe 'coordinate' (see issue #15)
        longitude: 13.4 // required
        latitude: 52.5 // required
    },

    // second proposal for coordinates (inline)
    longitude: 13.4, // optional
    latitude: 52.5 // optional
}
```

While the second proposal for `coordinates` reduces the depth of the data, the first variant embraces that both coordinate dimensions condition each other: There must always be a latitude if there is a longitude, and vice versa.

I propose a small github poll on this issue, vote 👍 for the first proposal (separate coordinates object) and ❤️ for the second variant (inline latitude, longitude).

**Comment by juliuste** [comment]
(If you have any objections, feel free to share them, don't be cowed by the poll)

**Comment by kiliankoe** [comment]
GitHub's mobile site still doesn't do emoji reactions :/

I much prefer the first variant to store the coordinate data in a separate object to stay in line with coordinate handling elsewhere in the spec.

**Comment by derhuerst** [comment]
Everywhere else, we would use `location` objects instead of `coordinates`.

**Comment by juliuste** [comment]
> I much prefer the first variant to store the coordinate data in a separate object to stay in line with coordinate handling elsewhere in the spec.

Please note though that we also proposr to replace the current `coordinates` and `address` attributes with the new `location` objects.

**Comment by kiliankoe** [comment]
Oh, must've read over that part^^

In that case I think I might actually prefer variant 2, although having the _optionality_ of one field being determined by the other is kinda weird (probably just programmatically, not really regarding the data itself), but that's just a minor detail. And it would "solve" #15 by just ignoring it altogether 🙈😄

**Comment by derhuerst** [comment]
So? What do you do now that we have a tie?

**Comment by juliuste** [comment]
Ok, to make some progress here, I'd be willing to withdraw my vote. So we inline the locations, which conveniently also closes #15, as @kiliankoe already pointed out.
